### PR TITLE
File-writer control changes

### PR DESCRIPTION
**Pull request by SkyToGround**
## Changes

* Added code for encoding/decoding the action response (_answ_) and writing done (_wrdn_) schemas.
* Added/modified code for handling the ``command_id`` field in the _stop-time_ schema.
* Added _metadata_ field code to start message encode/decoder.
* Changed code to use `from typing import NamedTuple` for decoders that I touched.
* Minor additions and fixes to the documentation.
* Minor changes done by **black**.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 setup(
     name="ess_streaming_data_types",
-    version="0.9.1",
+    version="0.9.3",
```
Comment (mattclarke) [inline setup.py:18]: 0.9.2?

File: README_DEV.md [hunk 2]
```diff
@@ -68,7 +81,7 @@ twine upload --repository-url https://test.pypi.org/legacy/ dist/*
 
 The new module can then be installed from test.pypi.org like so:
 ```
-pip uninstall ess_streaming_data_types    # Remove old version if present
+pip uninstall ess-streaming-data-types    # Remove old version if present
```
Comment (mattclarke) [inline README_DEV.md:84]: Good spot - thanks 👍

File: setup.py (hunk 1 above)
Comment (SkyToGround) [reply setup.py:18]: Right, my bad.
```suggestion
    version="0.9.2",
```

File: README_DEV.md [hunk 3]
```diff
@@ -1,6 +1,19 @@
 # Python Streaming Data Types
 ## For developers
 
+### Install local (development) version
+
+First, uninstall existing version of the Python streaming data types pacakge.
```
Comment (mattclarke) [inline README_DEV.md:6]:
```suggestion
First, uninstall existing version of the Python streaming data types package.
```

**Comment by SkyToGround** [comment]
@mattclarke Thanks for updating the README. Right now, it is only the file-writer that will be producing ``answ`` messages. Thus we do not currently have to worry about if the C++ code can verify Python produced ``answ`` messages.

File: streaming_data_types/finished_writing_wrdn.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+FILE_IDENTIFIER = b"wrdn"
+
+
+def serialise_wrdn(
+    service_id: str,
+    job_id: str,
+    error_encountered: bool,
+    file_name: str,
+    metadata: str = "",
+    message: str = "",
```
Comment (matthew-d-jones) [inline streaming_data_types/finished_writing_wrdn.py:16]: Minor point, but having the default value for `metadata` and `message` as `None` might make it clearer that these are omitted from the flatbuffer, which is different to empty strings being added to the flatbuffer.

**Review by matthew-d-jones — APPROVED** [review]
One minor suggestion to consider, but otherwise all looks well to me.

File: streaming_data_types/finished_writing_wrdn.py (hunk 4 above)
Comment (SkyToGround) [reply streaming_data_types/finished_writing_wrdn.py:16]: Sure. Will fix.

**Pull request merged**